Code/GraphMol/FileParsers/MaeMolSupplier.cpp

```diff
@@ -0,0 +1,144 @@
+//
+//  Copyright (C) ????  Who to put here when contributing to rdkit?
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+#include <iostream>
+#include <fstream>
+#include <map>
+#include <RDGeneral/BadFileException.h>
+#include <RDGeneral/FileParseException.h>
+#include <GraphMol/MolInterchange/details.h>
+#include <GraphMol/MolOps.h>
+#include <GraphMol/RWMol.h>
+#include <GraphMol/FileParsers/MolSupplier.h>
+#include <CoordGen/maeparser/Reader.hpp>
+
+
+namespace RDKit {
+
+using RDKit::MolInterchange::bolookup;
+
+MaeMolSupplier::MaeMolSupplier(std::istream *inStream, bool takeOwnership,
+                               bool sanitize, bool removeHs) {
+  dp_inStream = inStream;
+  df_owner = takeOwnership;
+  df_sanitize = sanitize;
+  df_removeHs = removeHs;
+
+  d_reader.reset(new schrodinger::mae::Reader(*inStream));
+  d_next_struct = d_reader->next("f_m_ct");
+}
+
+MaeMolSupplier::MaeMolSupplier(const std::string &fileName, bool sanitize, bool removeHs) {
+  auto *ifs = new std::ifstream(fileName.c_str(), std::ios_base::binary);
+  if (!ifs || !(*ifs) || ifs->bad()) {
+    std::ostringstream errout;
+    errout << "Bad input file " << fileName;
+    throw BadFileException(errout.str());
+  }
+  dp_inStream = (std::istream *)ifs;
+  df_owner = true;
+  df_sanitize = sanitize;
+  df_removeHs = removeHs;
+
+  d_reader.reset(new schrodinger::mae::Reader(*ifs));
+  d_next_struct = d_reader->next("f_m_ct");
```

Comment:
What would you think of putting these magic strings into the maeparser code and giving them names that could possibly be understood by third party users? Then they would serve as documentation, too.

I mean, these are almost certainly never going to change, but at the very least it would start to put an official record of the strings and their meanings in one place.

Reply:
Yeah that sounds reasonable to me.  Magic strings being owned by the maestro reader seems like the right way to do it.